src/components/pages/GarmentDetail.vue

```diff
@@ -90,44 +90,93 @@
-    //       this.dataIsLoaded = true;
-    //     })
-    //     .catch(error => this.showError(error.message));
-    // },
     data() {
       return {
-        garment: {},
+        garment: {
+          reference: '',
+          title: '',
+          description: '',
+          status: '',
+          type: '',
+          creator: '',
+          creation_date: '',
+          licence: 'Information not available yet on public API',
+          commitChanges: '',
+          contributors: '',
+          numberOfProposals: '',
+          images: [],
+          infos: [],
+          files: [],
+        },
         dataIsLoaded: false,
       };
     },
     methods: {
       showError(error) {
         EventBus.$emit('showError', error);
       },
+      formatRepoDetails(repoDetails) {
+        this.garment.creator = repoDetails.owner.login;
+        this.garment.creation_date = repoDetails.created_at;
+        this.garment.reference = repoDetails.id;
+        this.garment.description = repoDetails.description;
+        this.garment.infos.push({
+          label: 'favourites',
+          value: repoDetails.watchers,
+        });
+      },
+      formatRepoContributorStats(contributors) {
+        let totalCommits = 0;
+        contributors.forEach((contributor) => {
+          totalCommits += contributor.total;
+        });
+        this.garment.commitChanges = totalCommits;
+        this.garment.contributors = contributors.length;
+      },
+      formatRepoContents(repoContents) {
+        Object.assign(this.garment, repoContents);
+      },
+      formatRepoPullRequests(repoPullRequests) {
+        this.garment.numberOfProposals = repoPullRequests.length;
+      },
+      formatRepoReleases(repoReleases) {
+        repoReleases[0].assets.forEach((asset) => {
+          this.garment.files.push({
+            filetype: mime.extension(asset.content_type),
+            url: asset.url,
+            available: asset.state === 'uploaded',
+          });
+        });
+      },
     },
     filters: {
       moment: date => moment(date).format('L'),
     },
     props: ['user', 'repo'],
     mounted() {
-      const gh = new GitHub({
+      const gh = new Github({
         token: LoginStore.state.token,
       });
 
       const remoteRepo = gh.getRepo(this.user, this.repo);
 
-      remoteRepo.getContents('master', 'info.json', true, (err, content) => {
-        // console.log(content);
-        this.garment = content;
-        this.dataIsLoaded = true;
-        console.log(this.garment);
-      });
+      const repoDetails = remoteRepo.getDetails();
+      const repoContributorStats = remoteRepo.getContributorStats();
+      const repoContents = remoteRepo.getContents('master', 'info.json', true);
+      const repoPullRequests = remoteRepo.listPullRequests();
+      const repoReleases = remoteRepo.listReleases();
+
+      Promise.all([repoDetails, repoContributorStats, repoContents, repoPullRequests, repoReleases])
+        .then((response) => {
```

Comment:
optionally, you can use destructuring here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment